Tests/Opc.Ua.Gds.Tests/GlobalDiscoveryTestClient.cs

```diff
@@ -115,6 +117,59 @@ public async Task LoadClientConfiguration(int port = -1)
                 AppUser = new UserIdentity(gdsClientConfiguration.AppUserName, gdsClientConfiguration.AppPassword);
             }
             AdminUser = new UserIdentity(gdsClientConfiguration.AdminUserName, gdsClientConfiguration.AdminPassword);
+            Anonymous = new UserIdentity();
+        }
+        /// <summary>
+        /// Register the Test Client at the used GDS, needed to test the ApplicationSelfAdminPrivilege
+        /// </summary>
+        /// <param name="application"></param>
+        public bool RegisterTestClientAtGds()
+        {
+            //fill application record data type with own Data
+            OwnApplicationTestData = new ApplicationTestData {
+                ApplicationRecord = new ApplicationRecordDataType {
+                    ApplicationUri = m_client.Configuration.ApplicationUri,
+                    ApplicationType = m_client.Configuration.ApplicationType,
+                    ProductUri = m_client.Configuration.ProductUri,
+                    ApplicationNames = new LocalizedTextCollection() { new LocalizedText(m_client.Configuration.ApplicationName) },
+                    ApplicationId = new NodeId(Guid.NewGuid())
+                },
+                Subject = $"CN={m_client.Configuration.ApplicationName},DC=opc.tcp://localhost,O=OPC Foundation",
+            };
+
+            //connect
+            m_client.AdminCredentials = AdminUser;
+            m_client.Connect(m_client.EndpointUrl).Wait();
+            //register
+            var id = m_client.RegisterApplication(OwnApplicationTestData.ApplicationRecord);
+            if (id == null) return false;
+            OwnApplicationTestData.ApplicationRecord.ApplicationId = id;
+            //request new Cert
+            var req_id = m_client.StartNewKeyPairRequest(
+                OwnApplicationTestData.ApplicationRecord.ApplicationId,
+                OwnApplicationTestData.CertificateGroupId,
+                OwnApplicationTestData.CertificateTypeId,
+                OwnApplicationTestData.Subject,
+                OwnApplicationTestData.DomainNames,
+                OwnApplicationTestData.PrivateKeyFormat,
+                OwnApplicationTestData.PrivateKeyPassword
+                );
+
+            if (req_id == null) return false;
+            OwnApplicationTestData.CertificateRequestId = req_id;
+            //get cert
+            byte[] certificate = m_client.FinishRequest(
+                                OwnApplicationTestData.ApplicationRecord.ApplicationId,
+                                OwnApplicationTestData.CertificateRequestId,
+                                out _,
+                                out _
+                                );
+            if (certificate == null) return false;
+            //apply cert
+            m_client.Configuration.SecurityConfiguration.ApplicationCertificate = new CertificateIdentifier(certificate); //signed cert
```

Comment:
@mregen I started the implementation, however i run into some issues assinging the client the new gds signed certificate, where do I have to provide the private key.
In this line the configuration has the signed cert, and after the call to CheckApplicationInstanceCertificate
it reverts back to the selfsigned cert.

Reply:
![image](https://github.com/OPCFoundation/UA-.NETStandard/assets/7413710/4173c866-aba5-4df5-8814-d0e34b297070)

Reply:
Do i have to assign to the existing certificate identifier?

Reply:
![image](https://github.com/OPCFoundation/UA-.NETStandard/assets/7413710/ff55202e-2120-4ed7-a6ec-8daf7a8f7474)
fixed it, private key was missing